src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,251 @@
+      StringEntity stringEntity = new StringEntity(gson.toJson(payloadRequest), ContentType.APPLICATION_JSON);
+      requestBuilder.setEntity(stringEntity);
+      apiResponse = restApi.executePost(requestBuilder.build());
+
+      if (!apiResponse.isSuccess()) {
+        LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+        throw new RetryableException();
+      } else {
+        JsonObject responseJSON = jsonParser.parse(apiResponse.getResponseBody()).getAsJsonObject();
+        JsonArray servicedRequestsArray = responseJSON.get(ServiceNowConstants.SERVICED_REQUESTS).getAsJsonArray();
+        for (int i = 0; i < servicedRequestsArray.size(); i++) {
+          if (servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.STATUS_CODE)
+            .getAsInt() == HttpStatus.SC_FORBIDDEN) {
+            String encodedResponseBody = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.BODY)
+              .getAsString();
+            String decodedResponseBody = new String(Base64.getDecoder().decode(encodedResponseBody));
+            String errorDetail = jsonParser.parse(decodedResponseBody).getAsJsonObject().get(ServiceNowConstants.ERROR)
+              .getAsJsonObject().get(ServiceNowConstants.ERROR_DETAIL).getAsString();
+            if (errorDetail.equals(ServiceNowConstants.ACL_EXCEPTION)) {
+              throw new RuntimeException(String.format("Permission denied for '%s' operation.", config.getOperation()));
+            } else {
+              LOG.warn(String.format("'%s' operation failed for '%s'", config.getOperation(), config.getTableName()));
+              LOG.error("Error Response : {} ", decodedResponseBody);
+            }
+          }
+        }
+
+        JsonArray unservicedRequestsArray = responseJSON.get(ServiceNowConstants.UNSERVICED_REQUESTS).getAsJsonArray();
+        if (unservicedRequestsArray.size() > 0) {
+          retryUnservicedRequests(restRequestList, unservicedRequestsArray, accessToken);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error while connecting to ServiceNow", e.getMessage());
+      throw new RetryableException();
+    } catch (Exception e) {
+      throw e;
+    }
+    return apiResponse.getHttpStatus() == HttpStatus.SC_OK;
+  }
+
+  private String getErrorMessage(String responseBody) {
+    try {
+      JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
+      return jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString();
+    } catch (Exception e) {
+      return e.getMessage();
+    }
+  }
+
+  public ServiceNowBatchRequest getPayloadRequest(List<RestRequest> restRequests) {
+    ServiceNowBatchRequest payloadRequest = new ServiceNowBatchRequest();
+    payloadRequest.setBatchRequestId(batchRequestIdCounter.toString());
+    payloadRequest.setRestRequests(restRequests);
+
+    return payloadRequest;
```

Comment:
Please remove this extra space.

Reply:
removed. 659b6bd2dc503d409443ef1b43804566a4e25be0